### ISSUE TITLE
Old Issues

### ISSUE THREAD

**Issue by Iristyle** [Iristyle/Authentic#1]
- TODO: verify that rule violations properly propagate up and become `ConfigurationErrorsExceptions` -- look at `HttpContextInspectingAuthenticatorConfigurationElementValidator` / `HttpContextInspectingAuthenticatorConfigurationElement` for reference
- also look at the constructor of `EPS.Web.Authentication.Digest.AuthenticationInspectingAuthenticator` to see stuff that needs to be moved out into a separate validator
- TODO: flush out tests for `HttpContextInspectingAuthenticatorConfigurationElementValidatorTest`
- TODO: `PrincipalBuilderCache` doesnt' validate that types exist -- perhaps toss an `ArgumentException` -- probably makes sense
  http://fluentvalidation.codeplex.com/

*\* something like this **

``` csharp
    public class DigestHttpContextInspectingAuthenticatorConfigurationElementValidator :
        AbstractValidator<IAuthenticationHeaderInspectorConfigurationElement>
    {
        /// <summary>
        /// Initializes a new instance of the DigestHttpContextInspectingAuthenticatorConfigurationElementValidator class.
        /// </summary>
        public DigestHttpContextInspectingAuthenticatorConfigurationElementValidator()
        {
            new HttpContextInspectingAuthenticatorConfigurationElementValidator().r

        }
    }
```

**Comment by jarlah** [comment]
Going to port it to Scala. Objections? 

Edit. Saw the abandonware title now

**Comment by jarlah** [comment]
The point is that digest is still viable.. enough. Besides certificates it's the next best thing for authentication with small devices. If https used coupled with digest i suppose it's very difficult to break. It will need breaking of https protocol AND md5 by using rainbow tables. And the latter is almost Impossible with normal equipment

**Comment by jarlah** [comment]
Yes man in the middle. So the units must be strict with https validation;) which obviously isnt going to happen with small devices where cpu and ram is precious. But point still stands

**Comment by jarlah** [comment]
Done. I used some of the abstractions, loosely. And used most of the logic. But all in all a diffferent project.

https://github.com/jarlah/AuthenticScala

Works completely different too. Its independent of any form of framework conventions.
